subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc

```diff
@@ -69,21 +69,36 @@ But, because this package name doesn't contain the word `internal`, many Gradle
 Gradle 7.1 addresses the situation and marks the package as public.
 The classes that were unintentionally exposed are either deprecated or removed, depending on their external usage.
 
-The following classes are now officially recognized as public API:
+===== The following classes are now officially recognized as public API:
 - `GradleVersion`
 - `Path`
 - `Configurable`
 
-The following classes have known usages in external plugins and are now deprecated and set for removal in Gradle 8.0:
+===== The following classes have known usages in external plugins and are now deprecated and set for removal in Gradle 8.0:
 
 - `VersionNumber`
 - `TextUtil`
 - `WrapUtil`
 - `RelativePathUtil`
 - `DistributionLocator`
 - `SingleMessageLogger`
+- `ConfigureUtil`
 
-The following classes have only internal usages and were moved from `org.gradle.util` to the `org.gradle.util.internal` package:
+`ConfigureUtil` is used in plugins for applying configuration defined in a Groovy closure to DSL objects. This implementation should be avoided it will be bound to the Groovy. Instead, the object
```

Comment:
Since this duplicates mostly what we say in the javadoc, let's just link to the class javadoc for ConfigureUtil.  e.g., we can say a shortened version here with something like:

> `ConfigureUtil` is being removed without a replacement. Plugins can avoid the need for using `ConfigureUtil` by following [our example](link to javadoc).

Reply:
I've updated the upgrade guide.